lib/oxrdf/src/literal.rs

```diff
@@ -665,4 +726,102 @@ mod tests {
         assert_eq!("NaN", Literal::from(f32::NAN).value());
         assert_eq!("NaN", Literal::from(f64::NAN).value());
     }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_serde() {
+        // Simple literal
+        let j = serde_json::to_string(&Literal::new_simple_literal("foo")).unwrap();
+        assert_eq!("{\"value\":\"foo\"}", j);
+
+        let mut de = serde_json::Deserializer::from_str(&j);
+        let node: Literal = Deserialize::deserialize(&mut de).unwrap();
+        assert_eq!(node, Literal::new_simple_literal("foo"));
+
+        // Typed literal
+        let j = serde_json::to_string(&Literal::new_typed_literal("true", xsd::BOOLEAN)).unwrap();
+
+        assert_eq!(
+            "{\"value\":\"true\",\"datatype\":\"http://www.w3.org/2001/XMLSchema#boolean\"}",
+            j
+        );
+
+        let mut de = serde_json::Deserializer::from_str(&j);
+        let node: Literal = Deserialize::deserialize(&mut de).unwrap();
+        assert_eq!(node, Literal::new_typed_literal("true", xsd::BOOLEAN));
+
+        // Language-tagged string
+        let j = serde_json::to_string(&Literal::new_language_tagged_literal("foo", "en").unwrap())
+            .unwrap();
+        assert_eq!("{\"value\":\"foo\",\"language\":\"en\"}", j);
+        let mut de = serde_json::Deserializer::from_str(&j);
+        let node: Literal = Deserialize::deserialize(&mut de).unwrap();
+        assert_eq!(
+            node,
+            Literal::new_language_tagged_literal("foo", "en").unwrap()
+        );
+    }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn test_serde_from_reader() {
```

Comment:
Isn't it a given from Serde?

Reply:
Marking as resolved since we've had the `Deserialise` vs. `DeserialiseOwned` conversation in the other thread.